backend/src/middlewares/auth/auth.middleware.ts

```diff
@@ -32,4 +35,26 @@ const localAuth = (
   )(req, res, next);
 };
 
-export { localAuth };
+const signUpMiddleware = async (
+  req: TypedRequestBody<Prisma.UserCreateInput>,
+  res: Response,
+  next: NextFunction,
+): Promise<void> => {
+  try {
+    await signUpSchema.validate(req.body);
+    const existedUser = await prisma.user.findFirst({
+      where: {
+        email: req.body.email,
+      },
+    });
+    if (existedUser) throw new Error();
+
+    next();
+  } catch (err) {
+    console.error(err);
+    res.status(403);
```

Comment:
Use constants 
httpStatus.FORBIDDEN

Reply:
Yep, the hardcoded values mustn't be used in code. Or used only when it makes 100% sense to everyone else and it's just unavoidable under any circumstances

Reply:
Yes, thanks for the tip. Fixed it.